### PR TITLE
Cleanup test_other.py

### PR DESCRIPTION
**Pull request by sbc100**
- Remove use of builtin assert in favor of unittest assertions.
- Remove some loops to get more meaningful error message
- Remove some needless triple quotes

### CODE REVIEW

File: tests/test_other.py [hunk 1]
```diff
@@ -284,7 +284,6 @@ def test_emcc_2(self, compiler, suffix):
         print('skip because windows')
         continue
       target = args[1] if len(args) == 2 else 'hello_world.o'
-      self.clear()
```
Comment (kripken) [inline tests/test_other.py]: the clear guarantees we check that the emcc invocation emitted something, so I think it's worthwhile.

File: tests/test_other.py [hunk 2]
```diff
@@ -5251,7 +5253,7 @@ def test_locale_wrong(self):
     ''')
     run_process([PYTHON, EMCC, 'src.cpp', '-s', 'EXIT_RUNTIME=1', '-s', 'DISABLE_EXCEPTION_CATCHING=0'])
     self.assertContained('Constructed locale "C"\nThis locale is the global locale.\nThis locale is the C locale.', run_js('a.out.js', args=['C']))
-    self.assertContained('''Can't construct locale "waka": collate_byname<char>::collate_byname failed to construct for waka''', run_js('a.out.js', args=['waka'], assert_returncode=1))
+    self.assertContained('Can\'t construct locale "waka": collate_byname<char>::collate_byname failed to construct for waka', run_js('a.out.js', args=['waka'], assert_returncode=1))
```
Comment (kripken) [inline tests/test_other.py]: I think triple quotes are better than escaping, but I don't feel strongly, up to you.

Comment (sbc100) [reply tests/test_other.py]: Hmm.. maybe r"" is better then.   I usually reserve triple quotes for multi-line strings.

File: tests/test_other.py (hunk 1 above)
Comment (sbc100) [reply tests/test_other.py]: OK reverting.

**Pull request merged**